crypto/bio/bio_meth.c

```diff
@@ -63,12 +63,45 @@ int BIO_meth_set_write(BIO_METHOD *biom,
 
 int (*BIO_meth_get_read(BIO_METHOD *biom)) (BIO *, char *, int)
 {
+    return biom->bread_old;
+}
+
+int (*BIO_meth_get_read_ex(BIO_METHOD *biom)) (BIO *, char *, size_t, size_t *)
+{
     return biom->bread;
 }
 
+/* Conversion for old style bread to new style */
+int bread_conv(BIO *bio, char *out, size_t outl, size_t *read)
+{
+    int ret;
+
+    if (outl > INT_MAX)
+        return 0;
+
+    ret = bio->method->bread_old(bio, out, (int)outl);
+
+    if (ret <= 0) {
+        *read = 0;
+        return ret;
+    }
```

Comment:
Hmmm...I'm not sure I understand. If `ret == 0` then surely we need to return 0 (it's a failure). Changing to `if ( ret < 0)` would mean we would return 1 in this instance.